File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import math

def is_prime(num):
    if num<2:
        return False
    for j in range(2, int(math.sqrt(target))+1):
        if target % j == 0:
            return False
    return True

n = int(input())
arr = list(map(int,input().split()))
count = 0

for i in range(n):
    target = arr[i]
    if is_prime(target) == True:
        count += 1

print(count)

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        boolean[] arr = new boolean[n+1];
        for(int i=0; i<arr.length; i++){
            arr[i] = true;
        }
        
        arr[0] = arr[1] = false;
        
        for(int i=2; i<=Math.sqrt(n); i++){
            if(arr[i]){
                for(int j=i*i; j<=n; j+=i){
                    arr[j] = false;
                }
            }
        }
        
        for(boolean target : arr){
            if(target) answer++;
        }
        return answer;
    }
}